packages/@react-stately/collections/src/CollectionBuilder.ts

```diff
@@ -183,7 +188,7 @@ export class CollectionBuilder<T extends object> {
     }
 
     // Ignore invalid elements
-    if (partialNode.key == null) {
+    if (partialNode.key == null || partialNode.type == null) {
```

Comment:
was `type` being defined always a implied requirement here? It makes sense, but just wanna double check

Reply:
I think so? the type of the full node has these both as required, so I figure if we don't have them, then we have an invalid element